secure/storage/vault/src/lib.rs

```diff
@@ -513,6 +358,230 @@ impl Client {
     }
 }
 
+/// Processes a generic response returned by a vault request. This function simply just checks
+/// that the response was not an error and calls response.into_string() to clear the ureq stream.
+pub fn process_generic_response(resp: Response) -> Result<(), Error> {
```

Comment:
these kinda make me sad as it adds a bit more layers... perhaps instead we should make these methods on the structs themselves:

```
  from_response(resp: Response) -> Result<Self, Error> {
    ...
  }
```
thoughts?

Reply:
No worries! We'll take this offline and respond as appropriate in a follow up PR 😄